Analyses plots - remove WARNING statements for missing plots

PerryShafran-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

In a recent PR for analyses, it was noted that there was a WARNING for a plot that is missing due to lack of data needed to create a plot.  To remove the word WARNING, it was replaced with a Note and described that there is no plot due to lack of data to create a plot.  This often happens when there is a lack of data.  

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?

No

* Do you have any planned upcoming annual leave/PTO?

July 28-Aug 1
* Are there any changes needed in the times when the jobs are supposed to run/kick-off?

No
  
- [x ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x ] References the feature branch for `HOMEevs` are removed from the code.
- [x ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x ] Jobs over 15 minutes in runtime have restart capability.
- [x ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x ] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x ] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

1. git clone https://github.com/PerryShafran-NOAA/EVS.git
2. cd EVS; git checkout feature/analyses_missing_plot_fix
3. ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix
4. cd dev/drivers/scripts/plots/analyses
5. In the plot script, set the HOMEevs to your testing directory, and set COMIN to point to emc.vpppg.
6. Run the script using `qsub jevs_analyses_grid2obs_plots_last31days.sh`.  Use VDATE=20250708 for this test regardless of what day you test.

Commit:
Removing WARNING with a Note for any plots that are not generated

AliciaBentley-NOAA commented:
@PerryShafran-NOAA Can you please tell us a little more about why there "isn't data" to make these plots? The WARNING we saw previously said: `'WARNING: NO PLOT FOR,VIS,Guam,rtma'` Does RTMA not have visibility data over Guam, for example? If so, I think we should try to remove the verification that can't be done from EVS (so that it doesn't try to run at all on things it can't compute) rather than change the word WARNING to NOTE. 

CC @AndrewBenjamin-NOAA @sbanihash

PerryShafran-NOAA commented:
@AliciaBentley-NOAA It really depends on when the plot job is running.  For example, on the current plot job for today, there is a note made here:

`Note: No plot made due to lack of data,VIS,PuertoRico,rtma`

It's a different region than the previous warning, which is Guam.  But for today's plot job, there is no Note regarding any visibility data from Guam (all Guam plots exist).  

Also the issue in question usually occur for time series of individual thresholds.  For example, the plot is missing because it was a time series of visibility less than 0.5 mi.  I guess visibility less than 0.5 miles didn't occur over Puerto Rico in the last 31 days.  Thus, no plot.  That was likely the issue in Guam too, back in May.  

BTW, these individual time series for each threshold was a request last year.  I can't find the spreadsheet, but maybe these were chosen for removal?  These added to the number of plots produced and the time of the job from 1 hr to over 4 hr.

PerryShafran-NOAA commented:
Also, since I ran a test for VDATE=20250520 (the date where we originally saw the Guam warning), that Note/WARNING indeed popped up for visibility less than 0.5 mi (or 1 mi, for that plot job).  That same date didn't have warnings for Puerto Rico, though.  So it depends on situations where low visibility isn't recorded over a 31-day period.

AliciaBentley-NOAA commented:
@PerryShafran-NOAA Thanks! Can you please update your NOTEs to be a little more informative? Perhaps including information that thresholds not occurring in the last31days, etc.? As written, a reviewer/NCO SPA wouldn't know anything about they the data doesn't exist, just that the data doesn't exist and might be the result of an issue. Other components of EVS have messages about missing data because it did not meeting thresholds. Perhaps you could use those NOTEs/WARNINGs as an example? Thank you!

CC @AndrewBenjamin-NOAA @sbanihash

PerryShafran-NOAA commented:
@AliciaBentley-NOAA I'll do that!  

BTW, found the spreadsheet - I do not see any analysis plots that are scheduled for removal.

Commit:
Updated the Note description to be more detailed

PerryShafran-NOAA commented:
@AliciaBentley-NOAA I updated the note to say:

`Note: No plot made. The threshold does not occur over the region in the last 31 days`

And I also added the threshold in the description as well so folks know what threshold it is, the variable, and the region.

PerryShafran-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/analyses_missing_plot_fix

PerryShafran-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/analyses_missing_plot_fix

ShannonShields-NOAA commented:
@PerryShafran-NOAA I have started testing this PR; I will provide the job info tomorrow once the job is done running (it is a longer job and by the time it finishes, I will be on leave this afternoon).

ShannonShields-NOAA commented:
@PerryShafran-NOAA Here is the info for the analyses plot job. I noticed a few WARNINGs; are these to be expected?
```
WARNING: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/analyses/rtma.202507
01/evs.stats.rtma_anl.atmos.grid2obs.v20250701.stat does not exist
WARNING: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/analyses/rtma.202507
01/evs.stats.rtma_ges.atmos.grid2obs.v20250701.stat does not exist
```

`07/16 18:13:04.375 (time_series.py:481) WARNING: Could not find (and cannot plot) csi stats for VIS at any level. Continuing ...`
I do see instances of the following now:
`Note: No plot made. The threshold does not occur over the region in the last 31 days,VIS,PuertoRico,rtma,805`
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr784test/EVS/dev/drivers/scripts/plots/analyses/jevs_analyses_grid2obs_plots_last31days.o210807886
**DATA**: /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_analyses_grid2obs_plots_last31days.210807886.cbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/analyses/atmos.20250708
Tar file size:
`-rw-r--r-- 1 shannon.shields emc 1.4G Jul 16 18:38 evs.plots.analyses.atmos.grid2obs.last31days.v20250708.tar`

PerryShafran-NOAA commented:
@ShannonShields-NOAA The size of the tarball is good.

Yes, sometimes stats files are missing for whatever reason, but that doesn't affect the the plotting.  It goes on without these stats files.  Probably due to the switch.  

The Note is there as it should look like.

I do see the WARNING in the plot log that is included in the job log.  I don't think I noticed that before, but maybe I missed it or didn't address the plot code directly.  I could change that message so that there is something other than WARNING used here as well.  

I don't know if I should remove the WARNING for missing stat files.  @AndrewBenjamin-NOAA @AliciaBentley-NOAA Should I change this to a Note or something else?

ShannonShields-NOAA commented:
@PerryShafran-NOAA I think the WARNING about "stat does not exist" is fine, but we might want to ask @AliciaBentley-NOAA and @AndrewBenjamin-NOAA about the following WARNING:
`07/16 18:13:04.375 (time_series.py:481) WARNING: Could not find (and cannot plot) csi stats for VIS at any level. Continuing ...`
(especially because it has "cannot" which is a buzzword for NCO)

PerryShafran-NOAA commented:
For the WARNING about the csi stats, the word WARNING is automatic because the code is using the logging library within python and using the logger.warning to trigger this statement.  I could change this to logger.info and the word WARNING would no longer be there.  But because in these instances the word WARNING is fixed due to the python function it is using, I'd have to change the function to logger.info if we don't want the word WARNING.  

The words "Could not" and "cannot" can be changed by me easily if this is what we'd prefer.

@AliciaBentley-NOAA @AndrewBenjamin-NOAA @ShannonShields-NOAA

AndrewBenjamin-NOAA commented:
> For the WARNING about the csi stats, the word WARNING is automatic because the code is using the logging library within python and using the logger.warning to trigger this statement. I could change this to logger.info and the word WARNING would no longer be there. But because in these instances the word WARNING is fixed due to the python function it is using, I'd have to change the function to logger.info if we don't want the word WARNING.
> 
> The words "Could not" and "cannot" can be changed by me easily if this is what we'd prefer.
> 
> @AliciaBentley-NOAA @AndrewBenjamin-NOAA @ShannonShields-NOAA

Looking at the [NCEP WCOSS Implementation Standards Documentation](https://nws-hpc-standards.readthedocs.io/en/latest/), I think the CSI Warning is ok (non-fatal error), but should be reworded to remove the word "cannot". This should be an easy change as it looks like it is just part of an f-string within timeseries.py, you would just have to check to make sure to change it every time it appears.

@AliciaBentley-NOAA is this something that is already in EVSv1.0, and if yes is it worth fixing now if it is already in ops?

AliciaBentley-NOAA commented:
Hi all. This comment appears to be everywhere in EVS:
https://github.com/search?q=repo%3ANOAA-EMC%2FEVS+%22Could+not+find+%28and+cannot+plot%29%22&type=code

I don't think that we need to change everywhere, but we should try to prevent this WARNING from happening whenever possible. My question continues to be: What is different about "VIS,PuertoRico,rtma" that is causing this to happen? Looking at the [analyses Verification webpage](https://www.emc.ncep.noaa.gov/users/verification/analyses/prod/grid2obs/vis_rtma/), it looks like Puerto Rico is the only domain option in the dropdown menu that never has visibilities <1 or <0.5. Everywhere else does. How is that possible?

PerryShafran-NOAA commented:
> Hi all. This comment appears to be everywhere in EVS: https://github.com/search?q=repo%3ANOAA-EMC%2FEVS+%22Could+not+find+%28and+cannot+plot%29%22&type=code
> 
> I don't think that we need to change everywhere, but we should try to prevent this WARNING from happening whenever possible. My question continues to be: What is different about "VIS,PuertoRico,rtma" that is causing this to happen? Looking at the [analyses Verification webpage](https://www.emc.ncep.noaa.gov/users/verification/analyses/prod/grid2obs/vis_rtma/), it looks like Puerto Rico is the only domain option in the dropdown menu that never has visibilities <1 or <0.5. Everywhere else does. How is that possible?

When we did this test originally back in May (when you first noticed this WARNING), it was Guam that did not have a visibility <0.5 in the previous 31 days, but Puerto Rico did not have this error (I did run back for VDATE=20250520 for my own testing and only Guam VIS had this warning).  For the past 31 days from the VDATE of this test (20250708), it was Puerto Rico.  It appears that this remains the case now.  

I couldn't tell you why that is, but I don't think that there are a lot of observations in these locations, so I think it's possible.

PerryShafran-NOAA commented:
What I can do is, I think, change logger.warning to logger.info, to ensure that the word WARNING does not appear.  I think both these routines work in the same manner, so that should be an easy change.  In addition, I can reword the statement so the words "Could not" and "cannot" will be replaced by something else.  That should suffice for this PR.  If this sounds good, I can make this change.

AliciaBentley-NOAA commented:
@PerryShafran-NOAA A word like "unable" might be a good one to use instead of "cannot". So like: "Unable to find or plot csi stats for VIS at any level, possibly due to _________________________. Continuing..."

@ShannonShields-NOAA @AndrewBenjamin-NOAA

PerryShafran-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/analyses_missing_plot_fix

PerryShafran-NOAA commented:
> @PerryShafran-NOAA A word like "unable" might be a good one to use instead of "cannot". So like: "Unable to find or plot csi stats for VIS at any level, possibly due to _________________________. Continuing..."
> 
> @ShannonShields-NOAA @AndrewBenjamin-NOAA

I like that, I'll make that change!  Plus change to logger.info, to get rid of the word WARNING.  I'll need to test this before I change the repo.

PerryShafran-NOAA commented:
@AliciaBentley-NOAA @ShannonShields-NOAA The change to logger.info worked in removing the word WARNING from my latest analysis plot test.  The wording was also changed to Unable... per the suggestion of Alicia.  I think this is worth another test, so if you could go ahead and submit that, that would be great.

AliciaBentley-NOAA commented:
@PerryShafran-NOAA Did you push changes to your PR branch? I can't seem to find them.

PerryShafran-NOAA commented:
Whoops!  It's been a long weekend, I forgot I didn't push.  Doing so now.

Commit:
Updating logger.warning to logger.info and changing wording of information

PerryShafran-NOAA commented:
OK, I added the commit.  Weird things just happened as the changes disappeared after I tried to pull, but I got the fixes back in.  If you wish to check the wording of the information statement before you run, that would be fine.

ShannonShields-NOAA commented:
@AliciaBentley-NOAA and @AndrewBenjamin-NOAA Do you agree with the wording of the new logger statement for missing plots? I think it is fine and if you do too,  then I will kick off the job.

AliciaBentley-NOAA commented:
@PerryShafran-NOAA @ShannonShields-NOAA @AndrewBenjamin-NOAA The new messages look good to me! Thanks!

ShannonShields-NOAA commented:
@PerryShafran-NOAA The plot job has finished running. Please review the output; I now only noticed instances of WARNING due to "stat does not exist" which we expect from previous testing. I did see the fixed logger.info statements such as:
`07/21 15:17:55.630 (time_series.py:481) INFO: Unable to plot csi stats for VIS at any level due to lack of data. Continuing ...`
I did see 
```
nid002261: 2025-07-21T13:10:40.984007+00:00 nid002261 kernel: [15024081.616475
][T693440] mce: [Hardware Error]: Machine check events logged
```
but I remember seeing this before in one of your previous PRs and attributing it to the strangeness of Cactus and the prologue section (not part of the scripts), so I am ignoring it.
**Log**: /lfs/h2/emc/vpppg/noscrub/shannon.shields/pr784test/EVS/dev/drivers/scripts/plots/analyses/jevs_analyses_grid2obs_plots_last31days.o211371070
**DATA:** /lfs/h2/emc/stmp/shannon.shields/evs_test/prod/tmp/jevs_analyses_grid2obs_plots_last31days.211371070.cbqs01
**Plots**: /lfs/h2/emc/ptmp/shannon.shields/evs/v2.0/plots/analyses/atmos.20250708
Tar file size:
`-rw-r--r-- 1 shannon.shields emc 1.3G Jul 21 15:39 evs.plots.analyses.atmos.grid2obs.last31days.v20250708.tar`
Is this the same size as the parallel?

PerryShafran-NOAA commented:
@ShannonShields-NOAA Yeah, that error is part of the epilogue in this case.  I'm not entirely sure why things like this happens, but I'm not sure we can control that.

The size of the tarball is of expected size and looks good.  

The earlier Note statements are there as expected, as well as the new Unable to plot statement from time_series.py.  

I think that things look OK for this PR.

ShannonShields-NOAA reviewed (approved):
@PerryShafran-NOAA and @AndrewBenjamin-NOAA I have reviewed the script changes regarding statements about missing plots due to lack of data or threshold, and confirmed changes work as expected within analyses plot job. I approve this PR to be merged.

ShannonShields-NOAA commented:
@PerryShafran-NOAA Can you please double-check that your branch is synced?

PerryShafran-NOAA commented:
I always try to keep my codes up to date.  :)

We are good.

AndrewBenjamin-NOAA reviewed (approved):
I have reviewed the changes to the code and approve this PR
